src/app/course/token-use-snippet/token-use-snippet.component.ts

```diff
@@ -1,71 +1,115 @@
-import {Component, Inject, OnInit} from '@angular/core'
-import {CourseRegistration, User} from '@app/_models'
-import {TokenUseService} from '@app/course/_services/token-use.service'
+import {
+    AfterContentChecked,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+} from '@angular/core'
+import {Course, CourseEvent, User} from '@app/_models'
 import {ActivatedRoute} from '@angular/router'
-import {TokenUse} from '@app/_models/token_use'
 import {AuthenticationService} from '@app/_services/api/authentication'
-import {TuiNotification, TuiNotificationsService} from "@taiga-ui/core"
 import {CourseService} from "@app/course/_services/course.service"
+import {GradeBook} from "@app/_models/grade_book"
+
 
 @Component({
     selector: 'app-token-use-snippet',
     templateUrl: './token-use-snippet.component.html',
-    styleUrls: ['./token-use-snippet.component.scss']
+    styleUrls: ['./token-use-snippet.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class TokenUseSnippetComponent implements OnInit {
-    courseReg: CourseRegistration
+export class TokenUseSnippetComponent implements AfterContentChecked {
+    grades: GradeBook
+    gradesDisplayData: GradeBook
+    gradeBookTableHeaders: string[] = [
+        'name', 'event_name', 'grade'
+    ]
+    gradeBookTableDetailedHeaders: string[] = [
+        'name', 'event_name', 'grade', 'title', 'question_grade', 'attempts'
+    ]
+
+    // Filter
+    query: string
+    event: string
 
-    tokenUses: TokenUse[]
-    tokenUsesTableHeaders: string[] = ['assignment_name', 'tokens_required', 'points_given', 'maximum_number_of_use', 'actions']
+    // Pagination
+    numberOfGradeLines = 0
+    pageSize = 10
+    page = 0
 
```

Comment:
please remove the pagination.

Reply:
Can you test it with some large data. Let's try 500 students and 25 assignments.